src/test/java/ValidUrlsTest.java

```diff
@@ -0,0 +1,90 @@
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.TestInstance.Lifecycle.PER_CLASS;
+
+import io.jenkins.plugins.designlibrary.UISample;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Stream;
+import org.htmlunit.html.DomNode;
+import org.htmlunit.html.HtmlAnchor;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.jvnet.hudson.test.JenkinsRule;
+import org.jvnet.hudson.test.junit.jupiter.WithJenkins;
+
+@WithJenkins
+@TestInstance(PER_CLASS)
+class ValidUrlsTest {
+
+    private JenkinsRule jenkins;
+
+    private List<UISample> samples;
+
+    @BeforeAll
+    void beforeAll(JenkinsRule jenkins) {
+        this.jenkins = jenkins;
+        this.samples = jenkins.getInstance().getExtensionList(UISample.class);
+    }
+
+    /**
+     * Validate that relative URLs on a page actually link to a {@link UISample}
+     */
+    @ParameterizedTest
+    @MethodSource("getPages")
+    void validUrls(String url) throws Exception {
+        try (var webClient = jenkins.createWebClient().withJavaScriptEnabled(false)) {
+            // We get a bunch of spam in our logs about missing CSS, let's ignore that
+            webClient.getOptions().setPrintContentOnFailingStatusCode(false);
+            webClient.getOptions().setThrowExceptionOnFailingStatusCode(false);
+
+            List<String> validUrls =
+                    new ArrayList<>(samples.stream().map(UISample::getUrlName).toList());
+
+            var page = webClient.goTo(url);
+            var links = page.querySelectorAll(".jdl-section a");
+
+            if (links.isEmpty()) {
```

Comment:
I've never used them before but I think this would be a useful place to use assumptions from either [AssertJ](https://assertj.github.io/doc/#assertj-core-assumptions) or [Junit](https://junit.org/junit5/docs/current/user-guide/#writing-tests-assumptions)

Reply:
hmm I think not needed, the return is probably not needed either, as if links is empty, there's nothing to stream and it will just finish, the only reason its here is for the logging debug.

Reply:
Never seen that before, that's cool!

Gonna leave it as is as I quite like seeing that every page has been checked in IntelliJ :P + gonna keep the `return` just so it's clear we're not executing anything after.